crates/sui-core/src/consensus_types/committee_api.rs

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::consensus_types::AuthorityIndex;
+use narwhal_config::committee::AuthorityIdentifier;
+use sui_types::base_types::AuthorityName;
+use sui_types::committee::StakeUnit;
+
+pub(crate) trait CommitteeAPI {
+    fn total_stake(&self) -> StakeUnit;
+    fn authority_pubkey_by_index(&self, index: AuthorityIndex) -> Option<AuthorityName>;
+    fn authority_hostname_by_index(&self, index: AuthorityIndex) -> Option<&str>;
+    fn authority_stake_by_index(&self, index: AuthorityIndex) -> StakeUnit;
```

Comment:
I am thinking whether all these could be replaced/extracted under an `Authority` struct . Then the `CommitteeAPI` could have a method like `authority_by_index(&self, index: AuthorityIndex) -> Option<Authority>` . I assume now we want to avoid creating new structs all the time during conversion, but maybe something to note for the future.

Reply:
Until we merge Mysticeti repo into Sui repo, there is no way for Mysticeti to use Sui types. So I am avoiding using common types here (we would need to if we want it to return Authority). I am also avoiding clone of the host name which is a string.